network/p2p/middleware.go

```diff
@@ -177,7 +165,9 @@ func NewMiddleware(
 			ready()
 
 			<-ctx.Done()
-			mw.stop()
+
+			// wait for the readConnection and readSubscription routines to stop
+			mw.wg.Wait()
```

Comment:
I don't think we should to add a one-off timeout here since that's not a pattern we use anywhere else for shutdowns (afaik).

Components will all begin to shutdown in parallel when the node's context is cancelled, so worst case, the network hangs on shutdown while other components complete, and is eventually killed by the operator.

I'm not opposed to adding some shutdown timeout logic, but I think it should be added to the node builder framework.